src/App.tsx

```diff
@@ -1,24 +1,218 @@
+    todoId: number, completed: boolean,
+  ) => {
+    setTodoIdUpdate(state => [...state, todoId]);
+
+    try {
+      await updateTodo(todoId, { completed });
+
+      setTodos(state => state.map(todo => {
+        if (todo.id === todoId) {
+          return { ...todo, completed };
+        }
+
+        return todo;
+      }));
+    } catch {
+      setIsError(Error.UPDATE);
+    } finally {
+      setTodoIdUpdate(state => state.filter(todoItem => todoItem !== todoId));
+    }
+  }, [todos]);
+
+  const allCompleted = useMemo(() => (
+    todos.every(todo => todo.completed)
+  ), [todos]);
+
+  const toggleAllTodos = useCallback(async () => {
+    try {
+      await Promise.all(todos.map(todo => toggleTodo(todo.id, !allCompleted)));
+    } catch (error) {
+      setIsError(Error.UPDATE);
+    }
+  }, [todos, allCompleted]);
+
+  const changeName = async (
+    todoId: number, NewTitle: string,
+  ) => {
+    setTodoIdUpdate(state => [...state, todoId]);
+
+    try {
+      await updateTodo(todoId, { title: NewTitle });
+
+      setTodos(state => state.map(todo => {
+        if (todo.id === todoId) {
+          return { ...todo, title: NewTitle };
+        }
+
+        return todo;
+      }));
+    } catch {
+      setIsError(Error.UPDATE);
+    } finally {
+      setTodoIdUpdate(state => state.filter(todoItem => todoItem !== todoId));
+    }
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          todos={todos}
+          addTodo={addTodo}
+          setToggleAllTodos={toggleAllTodos}
+          isLoading={isLoading}
+        />
+        <TodoList
+          filteredTodos={filteredTodos}
+          removeTodo={removeTodo}
+          todoIdUpdate={todoIdUpdate}
+          tempTodo={tempTodo}
+          toggleCompletedTodo={toggleTodo}
+          changeName={changeName}
+        />
+        {!!todos.length && (
+          <Footer
+            todos={todos}
+            selectType={selectedType}
+            setSelectedType={setSelectedType}
+            removeCompletedTodos={removeCompletedTodos}
+          />
+        )}
+      </div>
+
+      <div className={classNames(
+        'notification is-danger is-light has-text-weight-normal',
+        { hidden: !isError },
+      )}
+      >
+        <button
+          type="button"
+          className="delete"
+          aria-label="DeleteButton"
+          onClick={() => setIsError(Error.NONE)}
+        >
+          x
+        </button>
+        {isError}
```

Comment:
- isError can be empty, so don't need to show empty space
- so, make sense to use condition here